clients/python/src/kaskada/client.py

```diff
@@ -159,9 +159,13 @@ def get_client(client: Optional[Client] = None) -> Client:
         Client: The provided client argument or the global client.
     """
     if client is None and KASKADA_DEFAULT_CLIENT is None:
-        raise ValueError("No client was provided, and no global client is configured. Consider setting the global client by creating a session, for example 'kaskada.api.session.LocalBuilder().build()`.")
+        raise ValueError(
+            "No client was provided, and no global client is configured. Consider setting the global client by creating a session, for example 'kaskada.api.session.LocalBuilder().build()`."
+        )
     elif client is None:
         client = KASKADA_DEFAULT_CLIENT
 
-    validate_client(client)
+    if client is not None:
+        validate_client(client)
+
```

Comment:
This will still error on mypy since the function returns `client` which is still `Optional` 

I used 

```
assert client is not None
```
and then after that line you can have 

```
validate_client(client)
return client
```